pkg/gadgets/internal/socketenricher/bpf/sockets-map.h

```diff
@@ -81,26 +81,38 @@ struct {
 static __always_inline struct sockets_value *
 gadget_socket_lookup(const struct __sk_buff *skb)
 {
-	// Only IPv4 is supported for now
-	if (load_half(skb, offsetof(struct ethhdr, h_proto)) != ETH_P_IP)
-		return 0;
-
 	struct sockets_key key = {0,};
+	int l4_off;
+
 	key.netns = current_netns;
-	key.family = AF_INET;
-	key.proto = load_byte(skb, ETH_HLEN + offsetof(struct iphdr, protocol));
+	key.family = load_half(skb, offsetof(struct ethhdr, h_proto));
+	switch (key.family) {
+	case ETH_P_IP:
+		key.family = AF_INET;
+		key.proto = load_byte(skb, ETH_HLEN + offsetof(struct iphdr, protocol));
+		l4_off = ETH_HLEN + sizeof(struct iphdr);
+		break;
+	case ETH_P_IPV6:
+		key.family = AF_INET6;
+		key.proto = load_byte(skb, ETH_HLEN + offsetof(struct ipv6hdr, nexthdr));
+		l4_off = ETH_HLEN + sizeof(struct ipv6hdr);
```

Comment:
Same here. However in this case it's not too bad because the current implementation just ignores those packets as nexthdr is not `IPPROTO_TCP` nor `IPPROTO_UDP`.